provider/fs/src/export/fs_exporter.rs

```diff
@@ -119,15 +119,19 @@ impl DataExporter for FilesystemExporter {
         locale: &DataLocale,
         obj: &DataPayload<ExportMarker>,
     ) -> Result<(), DataError> {
-        let mut path_buf = self.root.clone();
-        path_buf.push(&*key.write_to_string());
-        path_buf.push(&*locale.write_to_string());
-        path_buf.set_extension(self.manifest.file_extension);
-
-        if let Some(parent_dir) = path_buf.parent() {
-            fs::create_dir_all(parent_dir)
-                .map_err(|e| DataError::from(e).with_path_context(&parent_dir))?;
-        }
+        let mut path_buf = self.root.clone().into_os_string();
+        write!(
+            &mut path_buf,
+            "/{key}/{locale}.{}",
```

Comment:
Issue, here and elsewhere: using `/` is not good because Windows and other OSes need different separators. That is what PathBuf::push handles for us. I don't think either `write!` or `Path::new` fixes it for us since `write!` is writing directly to a string and `Path::new` tries to borrow where possible.

Reply:
Apparently this works because Windows tests are not failing. Note that before we used `push(&str)`, which used `AsRef<Path> for &str`, so the slashes in the `str` were still forward slashes (and `push` itself doesn't replace them).

Reply:
Okay, so before we were appending individual path segments for root, key, and locale, but the key can contain a slash which wasn't being processed.

I wouldn't be surprised if Windows accepts these characters but does some weird handling with them.

@eggrobin and @makotokato are the two ICU4X engineers who routinely use Windows and might be able to provide some insight here.